charts/operator/chart_test.go

```diff
@@ -0,0 +1,249 @@
+	).Funcs(
+		asciiStrs,
+		smallInts,
+		func(p *corev1.PullPolicy, c fuzz.Continue) {
+			policies := []corev1.PullPolicy{
+				corev1.PullAlways,
+				corev1.PullNever,
+				corev1.PullIfNotPresent,
+			}
+
+			*p = policies[c.Intn(len(policies))]
+		},
+		func(r corev1.ResourceList, c fuzz.Continue) {
+			r[corev1.ResourceCPU] = resource.MustParse(strconv.Itoa(c.Intn(1000)))
+			r[corev1.ResourceMemory] = resource.MustParse(strconv.Itoa(c.Intn(1000)))
+		},
+		//func(p *corev1.PodSpec, c fuzz.Continue) {
+		//	p.Containers = []corev1.Container{}
+		//},
+		func(p *corev1.Probe, c fuzz.Continue) {
+			p.InitialDelaySeconds = int32(c.Intn(1000))
+			p.PeriodSeconds = int32(c.Intn(1000))
+			p.TimeoutSeconds = int32(c.Intn(1000))
+			p.SuccessThreshold = int32(c.Intn(1000))
+			p.FailureThreshold = int32(c.Intn(1000))
+			p.TerminationGracePeriodSeconds = ptr.To(int64(c.Intn(1000)))
+		},
+		func(p *corev1.PodFSGroupChangePolicy, c fuzz.Continue) {
+			policies := []corev1.PodFSGroupChangePolicy{
+				corev1.FSGroupChangeOnRootMismatch,
+				corev1.FSGroupChangeAlways,
+			}
+
+			*p = policies[c.Intn(len(policies))]
+		},
+		func(s *intstr.IntOrString, c fuzz.Continue) {
+			*s = intstr.FromInt32(c.Int31())
+		},
+		func(s *corev1.ResourceName, c fuzz.Continue) { asciiStrs((*string)(s), c) },
+		func(_ *any, c fuzz.Continue) {},
+		func(_ *[]corev1.ResourceClaim, c fuzz.Continue) {},
+		func(_ *[]metav1.ManagedFieldsEntry, c fuzz.Continue) {},
+	)
+
+	schema, err := jsonschema.CompileString("", string(ValuesSchemaJSON))
+	require.NoError(t, err)
+
+	nilChance := float64(0.8)
+
+	files := make([]txtar.File, 0, 50)
+	for i := 0; i < 50; i++ {
+		// Every 5 iterations, decrease nil chance to ensure that we're biased
+		// towards exploring most cases.
+		if i%5 == 0 && nilChance > .1 {
+			nilChance -= .1
+		}
+
+		var values PartialValues
+		fuzzer.NilChance(nilChance).Fuzz(&values)
+		// Special case as fuzzer does not assign correctly scope
+		s := OperatorScope(Cluster)
+		values.Scope = &s
+		values.Webhook = &PartialWebhook{Enabled: ptr.To(true)}
+		makeSureTagIsNotEmptyString(values, fuzzer)
+
+		out, err := yaml.Marshal(values)
+		require.NoError(t, err)
+
+		merged, err := helm.MergeYAMLValues(t.TempDir(), DefaultValuesYAML, out)
+		require.NoError(t, err)
+
+		// Ensure that our generated values comply with the schema set by the chart.
+		if err := schema.Validate(merged); err != nil {
+			t.Logf("Generated invalid values; trying again...\n%v", err)
+			i--
+			continue
+		}
+
+		files = append(files, txtar.File{
+			Name: fmt.Sprintf("case-%03d", i),
+			Data: out,
+		})
+	}
+
+	for i := 0; i < 50; i++ {
+		// Every 5 iterations, decrease nil chance to ensure that we're biased
+		// towards exploring most cases.
+		if i%5 == 0 && nilChance > .1 {
+			nilChance -= .1
+		}
+
+		var values PartialValues
+		fuzzer.NilChance(nilChance).Fuzz(&values)
+		// Special case as fuzzer does not assign correctly scope
+		s := OperatorScope(Namespace)
```

Comment:
There is 50/51 split.
* [operator scope namespace](https://github.com/redpanda-data/helm-charts/pull/1509/files#diff-d7d89e287882d63ed9ce9a5e069524c6506f2e1e7669add7825429302cbe5a60R199)
* [operator scope cluster](https://github.com/redpanda-data/helm-charts/pull/1509/files#diff-d7d89e287882d63ed9ce9a5e069524c6506f2e1e7669add7825429302cbe5a60R165)

Reply:
OH. Not entirely sure how I missed that. Rather than copying this loop it would be much more clear to just wrap it in something like:
```go
for _, scope := range []OperatorScope{Namespace, Cluster} {

if scope == Namespace {
// Namespace specific code here.
}
}
```

Reply:
Right I will change that.